lib/elixir/lib/access.ex

```diff
@@ -137,7 +137,8 @@ defmodule Access do
 
   """
 
-  @type t :: list | map | nil | any
+  @type container :: keyword | struct | map | any
+  @type t :: container | nil
```

Comment:
I was forced to separate it as I couldn't make Dialyzer not complain in `Kernel.pop_in/2`(https://github.com/elixir-lang/elixir/pull/5991/files#diff-e06d5ebc2907d951188e88476f85af51R1974).
It's a bit annoying having the `nil` here as most of the times `nil` will raise so we have to manually create the spec for the function